vscode/src/client.ts

```diff
@@ -339,8 +339,29 @@ export default class Client extends LanguageClient implements ClientInterface {
         await vscode.window.showErrorMessage(
           `Ruby LSP error ${error.data.errorClass}: ${error.data.errorMessage}\n\n${error.data.backtrace}`,
         );
-      } else {
-        this.telemetry.logError(error, { ...error.data });
+      } else if (
+        error.data.errorMessage &&
+        error.data.errorClass &&
+        error.data.backtrace
+      ) {
+        // Sanitize the backtrace coming from the server to remove the user's home directory from it, then mark it as a
+        // trusted value. Otherwise the VS Code telemetry logger redacts the entire backtrace and we are unable to see
+        // where in the server the error occurred
+        const stack = new vscode.TelemetryTrustedValue(
+          error.data.backtrace
+            .split("\n")
+            .map((line: string) => line.replace(os.homedir(), "~"))
```

Comment:
(I was crious if `~` made sense on Windows, [seems it is on Powershell](https://superuser.com/questions/332871/what-is-the-equivalent-of-linuxs-tilde-in-windows), so this is fine).